Homework5/Routers/Routers/RoutersNetwork.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System.Text;
+
+namespace Routers;
+
+/// <summary>
+/// Represents routers network
+/// </summary>
+public class RoutersNetwork
+{
+    private readonly List<Edge> _edges = new ();
+    private readonly Heap<Edge> _availableEdges = new();
+    private readonly int _verticesCount;
+    
+    // я не придумал, как тестировать по-другому
+    public readonly Dictionary<Edge, int> TreeEdges = new();
+    public RoutersNetwork(string inputPath)
+    {
+        var lines = File.ReadAllLines(inputPath);
+        foreach (var line in lines)
+        {
+            var splitLine = line.Split();
+            var vertex0 = int.Parse(splitLine[0][..^1]);
+            var vertex1 = 0;
+            foreach (var s in splitLine)
+            {
+                // если вершина
+                if (int.TryParse(s, out var number))
+                {
+                    vertex1 = number;
+                    if (number > _verticesCount)
+                        _verticesCount = number;
+                } // если вес ребра
+                else if (s[0] == '(')
+                {
+                    var value = int.Parse(s[1..s.LastIndexOf(')')]);
+                    _edges.Add(new Edge(vertex0, vertex1, weight: value));
+                }
+            }
+        }
+    }
+    
+    /// <summary>
+    /// Writes optimal routers network in file
+    /// </summary>
+    public int BuildNetwork(string targetPath)
+    {
+        var processedVertices = new bool[_verticesCount + 1]; 
+        var outputLines = new StringBuilder[_verticesCount];
+        FillOutputPattern(outputLines);
+        UpdateAvailableEdges(1, processedVertices);
+        processedVertices[1] = true;
+        for (var i = 0;  i < _verticesCount - 1; i++)
+        {
+            var newTreeEdge = _availableEdges.Extract();
+            //  из _availableEdges нельзя просто так взять максимальное ребро, нужно еще раз проверить,
+            // что 1 вершина помечена, а другая нет
+            while (processedVertices[newTreeEdge.Vertex0] && processedVertices[newTreeEdge.Vertex1])
+            {
+                newTreeEdge = _availableEdges.Extract();
+            }
+            
+            // если не нашлось подходящего ребра но мы все ещё в цикле, то граф был несвязный
+            if (newTreeEdge.Weight == 0)
+            {
+                var ew = Console.Error;
+                ew.Write("routers network were unconnected");
```

Comment:
```suggestion
                Console.Error.Write("routers network were unconnected");
```
А то какое-то странное имя переменной